block/deserialize_test.go

```diff
@@ -0,0 +1,173 @@
+package block
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/vulpemventures/go-elements/network"
+	"github.com/vulpemventures/go-elements/payment"
+
+	"github.com/vulpemventures/go-elements/elementsutil"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBlockDeserialization(t *testing.T) {
+	file, err := ioutil.ReadFile("testdata/deserialize.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	var tests []map[string]interface{}
+	err = json.Unmarshal(file, &tests)
+
+	for _, v := range tests {
+		testName := v["name"].(string)
+		t.Run(testName, func(t *testing.T) {
+			block, err := NewFromHex(v["hex"].(string))
+			if err != nil {
+				t.Errorf("test: %v, err: %v", testName, err)
+			}
+
+			assert.Equal(
+				t,
+				v["numOfTx"].(string),
+				strconv.Itoa(len(block.TransactionsData.Transactions)),
+			)
+		})
+	}
+}
+
+func TestBlockDeserializationIntegration(t *testing.T) {
```

Comment:
I think this might be overkill.
I mean, IMO it's enough to have fixtures with dynafed/non-dynafed blocks and make sure that they are correctly deserialized. I don't see any added value in creating a tx, broadcast it and deserialize the raw block fetched from some explorer. This just adds some (most likely a lot) overhead to the test, but basically checks the same thing of `TestDeserialization`.

Reply:
I just wanted to make sure that beside header also transactions are deserialised correctly.

Reply:
couldn't we use fixtures also for this? By the way, I thought this was already ensured by the test above

Reply:
i wanted to check value if it is correctly deserialized, to have live integration test, in fixtures i used, i would need to have blinding keys which i dont, you think we should remove it?